### PR TITLE
KTOR-5688 Make NSURLSessionDelegate constructor public

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

**Review by marychatte — APPROVED** [review]
lgtm

**Review by rsinukov — CHANGES_REQUESTED** [review]
I think it's ok to make constructor public and delete factory functions. But I would prefer to keep everything else internal or private

File: ktor-client/ktor-client-darwin-legacy/darwin/src/io/ktor/client/engine/darwin/KtorLegacyNSURLSessionDelegate.kt [hunk 1]
```diff
@@ -35,11 +34,12 @@ public fun KtorLegacyNSURLSessionDelegate(): KtorLegacyNSURLSessionDelegate {
  *   * URLSession:task:willPerformHTTPRedirection:newRequest:completionHandler:
  */
 @OptIn(UnsafeNumber::class)
-public class KtorLegacyNSURLSessionDelegate internal constructor(
-    private val challengeHandler: ChallengeHandler?
+public class KtorLegacyNSURLSessionDelegate(
```
Comment (rsinukov) [inline ktor-client/ktor-client-darwin-legacy/darwin/src/io/ktor/client/engine/darwin/KtorLegacyNSURLSessionDelegate.kt:37]: factory function can be removed, if you add default parameter

File: ktor-client/ktor-client-darwin-legacy/darwin/src/io/ktor/client/engine/darwin/internal/legacy/DarwinLegacyTaskHandler.kt [hunk 2]
```diff
@@ -17,11 +16,11 @@ import platform.Foundation.*
 import kotlin.coroutines.*
 
 @OptIn(DelicateCoroutinesApi::class)
-internal class DarwinLegacyTaskHandler(
+public class DarwinLegacyTaskHandler(
```
Comment (rsinukov) [inline ktor-client/ktor-client-darwin-legacy/darwin/src/io/ktor/client/engine/darwin/internal/legacy/DarwinLegacyTaskHandler.kt:19]: why do other classes need to be public?

**Review by rsinukov — APPROVED** [review]
LGTM, but minot comments

File: ktor-client/ktor-client-darwin-legacy/darwin/src/io/ktor/client/engine/darwin/internal/legacy/DarwinLegacyTaskHandler.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-darwin-legacy/darwin/src/io/ktor/client/engine/darwin/internal/legacy/DarwinLegacyTaskHandler.kt]: thanks, fixed

**Pull request merged**